packages/eslint-plugin/tests/rules/require-await.test.ts

```diff
@@ -134,14 +134,6 @@ async function testFunction(): Promise<void> {
       async value => Promise.resolve(value),
     ),
   );
-}
-    `,
-    'async function* run() {}',
```

Comment:
I reviewed the valid test cases from ESLint and found that some of them were missing. This was one of them:
https://github.com/eslint/eslint/blob/0dd38631b8dd60f93807e9cee1df3e752b86ab51/tests/lib/rules/require-await.js#L77

...as was the test below:
https://github.com/eslint/eslint/blob/0dd38631b8dd60f93807e9cee1df3e752b86ab51/tests/lib/rules/require-await.js#L68-L72

...and the one a bit further down on line 217 (except that one had a return type annotation:
https://github.com/eslint/eslint/blob/0dd38631b8dd60f93807e9cee1df3e752b86ab51/tests/lib/rules/require-await.js#L81

I've moved them down into the "base eslint tests" section.